nameko_grpc/connection.py

```diff
@@ -331,7 +378,14 @@ def send_request(self, request_headers):
         over the response.
 
         Invocations are queued and sent on the next iteration of the event loop.
+
+        raises ConnectionTerminatingError if connection is terminating. Check
+         connection .is_alive() before initiating send_request
         """
+        if self.terminating:
```

Comment:
This is exceptional behaviour. We are checking is_alive and initiating a new connection if the connection is terminating.